In the last few exercises you've learned everything there is to know about heights and weights of baseball players. Now it's time to dive into another sport: soccer.

You've contacted FIFA for some data and they handed you two lists. The lists are the following:

In [1]:
import pandas as pd
# Import numpy
import numpy as np
df_fifa = pd.read_csv('CSV/fifa.csv', index_col = 0)
heights = df_fifa['height'].tolist()
positions = df_fifa['position'].tolist()
print('positions =', positions[:5])
print('heights =', heights[:5])

positions = ['GK', 'M', 'A', 'D', 'M']
heights = [191, 184, 185, 180, 181]


Each element in the lists corresponds to a player. The first list, `positions`, contains strings representing each player's position. The possible positions are: `'GK'` (goalkeeper), `'M'` (midfield), `'A'` (attack) and `'D'` (defense). The second list, `heights`, contains integers representing the height of the player in cm. The first player in the lists is a goalkeeper and is pretty tall (191 cm).

You're fairly confident that the median height of goalkeepers is higher than that of other players on the soccer field. Some of your friends don't believe you, so you are determined to show them using the data you received from FIFA and your newly acquired Python skills.

**INSTRUCTIONS**

- Convert `heights` and `positions`, which are regular lists, to numpy arrays. Call them `np_heights` and `np_positions`.
- Extract all the heights of the goalkeepers. You can use a little trick here: use `np_positions == 'GK'` as an index for `np_heights`. Assign the result to `gk_heights`.
- Extract all the heights of all the other players. This time use `np_positions != 'GK'` as an index for `np_heights`. Assign the result to `other_heights`.
- Print out the median height of the goalkeepers using `np.median()`. Replace `None` with the correct code.
- Do the same for the other players. Print out their median height. Replace `None` with the correct code.

In [2]:


# Convert positions and heights to numpy arrays: print(np_heights), np_heights
np_heights = np.array(heights)
np_positions = np.array(positions)

# Heights of the goalkeepers: gk_heights
gk_heights = np_heights[np_positions == 'GK']
#print(gk_heights)

# Heights of the other players: other_heights
other_heights = np_heights[np_positions != 'GK']
#print(other_heights)

# Print out the median height of goalkeepers. Replace 'None'
gk_median = np.median(gk_heights)
print("Median height of goalkeepers: " + str(gk_median))

# Print out the median height of other players. Replace 'None'
other_median = np.median(other_heights)
print("Median height of other players: " + str(other_median))

Median height of goalkeepers: 188.0
Median height of other players: 181.0


Знайти добуток квадратів суми зростів нападаючих нижчих 180см і медіани захисників вищих 175см

In [27]:
np_heights = np.array(heights)
np_positions = np.array(positions)

a_heights = np_heights[np_positions == 'A']
#print(a_heights)

#A_heights = (a_heights[a_heights < 180])
A_heights = np.sum(a_heights[a_heights < 180])
print(A_heights ** 2)

d_heights = np_heights[np_positions == 'D']
D_heights = d_heights[d_heights > 175]
DM_heights = np.median(D_heights)
print(A_heights ** 2 * DM)

21516195856
184.0
